src/Npgsql/NpgsqlParameter.cs

```diff
@@ -519,11 +521,11 @@ internal void SetResolutionInfo(PgTypeInfo typeInfo, PgConverter converter, PgTy
     /// Attempt to resolve a type info based on available (postgres) type information on the parameter.
     internal void ResolveTypeInfo(PgSerializerOptions options)
     {
-        var previouslyResolved = TypeInfo?.Options == options;
+        var previouslyResolved = ReferenceEquals(TypeInfo?.Options, options);
         if (!previouslyResolved)
         {
             var staticValueType = StaticValueType;
-            var valueType = GetValueType(StaticValueType);
+            var valueType = GetValueType(staticValueType);
```

Comment:
We don't actually need to call `GetValueType` unless staticValueType is `object`, so we can move that all behind `staticValueType == typeof(object)` check (and we can get rid of `staticValueType` variable and instead just set `StaticValueType` to `valueType`).

Reply:
```
    Type? GetValueType(Type staticValueType) => staticValueType != typeof(object) ? staticValueType : Value?.GetType();
```
That is exactly what it should do, I assume it'll be inlined.

Reply:
It is indeed inlined, but we don't even have to call it unless we already know `staticValueType` is object. What I propose is:

```csharp
var valueType = StaticValueType;

string? dataTypeName = null;
DataTypeName? builtinDataTypeName = null;
...

if (valueType == typeof(object))
{
	valueType = GetValueType(valueType);
	if (valueType is null && pgTypeId is null)
...
```

Reply:
OK makes sense, will push